optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -33,14 +32,14 @@
     AutoModelForSeq2SeqLM,
     AutoModelForSpeechSeq2Seq,
     AutoModelForVision2Seq,
+    EncoderDecoderCache,
     GenerationConfig,
     Pix2StructForConditionalGeneration,  # Pix2struct does not support AutoModel
 )
 from transformers.file_utils import add_end_docstrings, add_start_docstrings_to_model_forward
-from transformers.generation.logits_process import WhisperTimeStampLogitsProcessor
 from transformers.modeling_outputs import BaseModelOutput, Seq2SeqLMOutput
 from transformers.models.auto.modeling_auto import MODEL_FOR_SPEECH_SEQ_2_SEQ_MAPPING_NAMES
-from transformers.models.whisper.tokenization_whisper import TASK_IDS, TO_LANGUAGE_CODE
+from transformers.models.whisper.generation_whisper import WhisperGenerationMixin
```

Comment:
this will break for transformers < v4.37.0 no ?

Reply:
should we update the minimum version for whisper ? because this way we are using the transformers custom generation loop, which solves many discrepancies with transformers transcription (e.g. 2 init tokens)

Reply:
for example if I run this code on main optimum:
```python
transformers_outputs = model.generate(
    **inputs,
    num_beams=1,
    use_cache=True,
    max_new_tokens=3,
    min_new_tokens=3,
)
print(transformers_outputs)

ort_outputs = ort_model.generate(
    **inputs,
    num_beams=1,
    use_cache=True,
    max_new_tokens=3,
    min_new_tokens=3,
)
print(ort_outputs)
```
I get:
```
tensor([[50257, 50362,   345,   345,   345],
        [50257, 50362,   345,   345,   345]])

tensor([[50257, 50362,   345,   345],
        [50257, 50362,   345,   345]])
```
and there other stuff that were added to the generation loop like short vs long form generation (probably related to https://github.com/huggingface/optimum/issues/1972)

Reply:
I see, would make sense to add it as done for decoder models
https://github.com/huggingface/optimum/blob/2a6d8573301b164155e6151ee60efd9ccf57554c/tests/onnxruntime/test_modeling.py#L2425